core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +314,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput' &&
+                  maxSelectionCount === 1) {
+        var numChoices = $scope.getAnswerChoices().length;
+        choiceRules = [];
```

Comment:
choiceRules was an attempt to follow your suggestion of having an array of booleans. So choiceRules gets the number of answer choices and starts as an array of false booleans. As each answer choice is checked for coverage, then the appropriate array index is turned to true.

In testing, I found that I needed BOTH the array of booleans checking the choice rules and the previous array that tested for whether each answer choice text appeared in order to get an accurate picture of whether an answer choice had in fact been handled. It may be that with the extra validation, this is no longer necessary.